core/Application/Kernel/EnvironmentValidator.php

```diff
@@ -0,0 +1,83 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Application\Kernel;
+
+use Piwik\Application\Kernel\GlobalSettingsProvider;
+use Piwik\Common;
+use Piwik\Piwik;
+use Piwik\SettingsServer;
+use Piwik\Translate;
+use Piwik\Translation\Translator;
+
+/**
+ * Validates the Piwik environment. This includes making sure the required config files
+ * are present, and triggering the correct behaviour if otherwise.
+ */
+class EnvironmentValidator
+{
+    /**
+     * @var GlobalSettingsProvider
+     */
+    protected $settingsProvider;
+
+    /**
+     * @var Translator
+     */
+    protected $translator;
+
+    public function __construct(GlobalSettingsProvider $settingsProvider, Translator $translator)
+    {
+        $this->settingsProvider = $settingsProvider;
+        $this->translator = $translator;
+    }
+
+    public function validate()
+    {
+        $inTrackerRequest = SettingsServer::isTrackerApiRequest();
+        $inConsole = Common::isPhpCliMode();
+
+        $this->checkConfigFileExists($this->settingsProvider->getPathGlobal());
+        $this->checkConfigFileExists($this->settingsProvider->getPathLocal(), $startInstaller = !$inTrackerRequest && !$inConsole);
+    }
+
+    /**
+     * @param $path
+     * @param bool $startInstaller
+     * @throws \Exception
+     */
+    private function checkConfigFileExists($path, $startInstaller = false)
+    {
+        if (is_readable($path)) {
+            return;
+        }
+
+        Translate::loadAllTranslations();
```

Comment:
Are you sure this is needed? I checked out the branch to give it a try and commenting this line doesn't *seem* to change anything, and it seems to make sense because the local (or global) config doesn't exist so…